File: Src/Icm.Core/EventArgs/EventArgs.vb
Namespace Icm

    Public Class EventArgs(Of T)
        Inherits EventArgs

        Private ReadOnly data_ As T
        ReadOnly Property Data() As T
            Get
                Return data_
            End Get
        End Property

        Public Sub New(ByVal _data As T)
            data_ = _data
        End Sub

    End Class

End Namespace

File: Src/Icm.Core/EventArgs/EventArgs.cs
using System;

namespace Icm
{

	public class EventArgs<T> : EventArgs
	{

		private readonly T data_;
		public T Data {
			get { return data_; }
		}

		public EventArgs(T _data)
		{
			data_ = _data;
		}

	}

}

//=======================================================
//Service provided by Telerik (www.telerik.com)
//Conversion powered by NRefactory.
//Twitter: @telerik
//Facebook: facebook.com/telerik
//=======================================================
